File: content/problems/1946-largest-number-after-mutating-substring/solutions/first-gain-contiguous-mutation/Solution.java
class Solution {
    public String maximumNumber(String num,int[] change){
        char[] result=num.toCharArray();
        boolean started=false;
        for(int i=0;i<result.length;i++){
            int digit=result[i]-'0',mapped=change[digit];
            if(!started){
                if(mapped>digit){
                    started=true;
                    result[i]=(char)('0'+mapped);
                }
            }else if(mapped>=digit)result[i]=(char)('0'+mapped);
            else break;
        }
        return new String(result);
    }
}

File: content/problems/1946-largest-number-after-mutating-substring/solutions/first-gain-contiguous-mutation/solution.cpp
class Solution {
    public: string maximumNumber(string num,vector<int>& change){
        bool started=false;
        for(char& value:num){
            int digit=value-'0',mapped=change[digit];
            if(!started){
                if(mapped>digit){
                    started=true;
                    value='0'+mapped;
                }
            }else if(mapped>=digit)value='0'+mapped;
            else break;
        }
        return num;
    }
};
